hibernate-core/src/main/java/org/hibernate/hql/spi/id/MultiTableBulkIdStrategy.java

```diff
@@ -23,6 +24,27 @@
  * @author Steve Ebersole
  */
 public interface MultiTableBulkIdStrategy {
+
+	/**
+	 * Prepare the strategy.  Called as the SessionFactory is being built.  Intended patterns here include:<ul>
+	 *     <li>Adding tables to the passed Mappings, to be picked by by "schema management tools"</li>
+	 *     <li>Manually creating the tables immediately through the passed JDBC Connection access</li>
+	 * </ul>
+	 * @param jdbcServices The JdbcService object
+	 * @param connectionAccess Access to the JDBC Connection
+	 * @param metadata Access to the O/RM mapping information
+	 * @param sessionFactoryOptions
+	 * @deprecated Will be removed in favor of the variant accepting a {@link SqlStringGenerationContext}
+	 * @see #prepare(JdbcServices, JdbcConnectionAccess, MetadataImplementor, SessionFactoryOptions, SqlStringGenerationContext)
+	 */
+	@Deprecated
+	default void prepare(JdbcServices jdbcServices, JdbcConnectionAccess connectionAccess, MetadataImplementor metadata, SessionFactoryOptions sessionFactoryOptions) {
```

Comment:
Just a warning: you're restoring backwards compatibility for callers of this method, but not for implementors. This method will never get called by Hibernate ORM.